_posts/2017-08-24-api-platform-en.md

```diff
@@ -0,0 +1,121 @@
+You can play with the interface and the auto-generated documentation before we go on and see another feature used many times in API: filters.
+
+## Filters and sorts
+
+Wouldn't it be nice to filter our movies by name, creation date, or any other property? Let's see how we can do that, in a very simple manner.
+
+The first thing is to declare a _symfony_ service detailing the way we want our properties to be filtered:
+
+```yaml
+services:
+    movies.search_filter:
+        parent:    'api_platform.doctrine.orm.search_filter'
+        arguments: [ { id: 'exact', name: 'partial', countryOfOrigin: 'partial' } ]
+        tags:      [ { name: 'api_platform.filter', id: 'movies.search_filter' } ]
+```
+
+As you can see, we define for all properties the level of comparison: for the _id_, we want an _exact_ match (id=5), for the name a _partial_ level, which is a LIKE %name%, and the same thing for our _countryOfOrigin_. Other levels are available to strings, but not only, you have numeric values, booleans, dates...
+
+Filter management happens automatically with Api Platform according to the type (_bool, integer…_), and it's very easy to filter values using operators `<=` and others.
+
+```yaml
+services:
+    movies.order_filter:
+        parent:    'api_platform.doctrine.orm.order_filter'
+        arguments: [ { id: ~, name: ~ } ]
+        tags:      [ { name: 'api_platform.filter', id: 'movies.order_filter' } ]
+```
+Above, we declare a service that allows to sort the fields _id_ and _name_.
+
+Now that our services are created, we only have to tell our Movie entity to use them:
+
+```php
+<?php
+/**
+ * A movie.
+ *
+ * @see http://schema.org/Movie Documentation on Schema.org
+ *
+ * @ORM\Entity
+ * @ApiResource(iri="http://schema.org/Movie", attributes={"filters"={"movies.search_filter", "movies.order_filter"}})
+ */
+class Movie
+{
+// ...
+}
+```
+You can now query your API with the following URL: `movies?name=O&order[name]=desc`  
+Very simple to implement right?
+
+Obviously, you can create your own [Api Platform filters](https://api-platform.com/docs/core/filters#creating-custom-filters).
+
+Many other features exist like :
+* [pagination](https://api-platform.com/docs/core/pagination)
+* a rich event system
+* an cache invalidation system (coming in version 2.1)
+
+## Bottom line
+
+Api Platform keeps its promise concerning how fast we can develop an API: creating an API in a few minutes is a child's play!
+
+Developed using best practices, API Platform is easily extensible, which allows everyone to add behaviour.
+
+Finally, the documentation is a very nice thing. You have plenty of examples and explanations.
+
+## To come
+
+Version 2.1 of API platform will be soon released, and comes with new features such as: an admin system developed in React, new filters... Details [here](https://dunglas.fr/2017/06/api-platform-2-1-feature-walkthrough-create-blazing-fast-hypermedia-apis-generate-js-apps/)
```

Comment:
`will be soon released` -> `will soon be released`? Not sure but it sounds better